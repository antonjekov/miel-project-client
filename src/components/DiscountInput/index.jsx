import React from 'react'
import styles from './index.module.css';
import { Form } from 'react-bootstrap';

function DiscountInput({ values, errors, handleChange }) {
    return (
        <Form.Group >
            <Form.Label>Discount</Form.Label>
            <Form.Control className={styles['form-control']} type="number" name='discount' value={values.discount} onChange={handleChange} isInvalid={!!errors.discount} isValid={values.discount && !errors.discount} />
            <Form.Control.Feedback type='invalid'>{errors.discount}</Form.Control.Feedback>
            <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
        </Form.Group>
    )
}

export default DiscountInput