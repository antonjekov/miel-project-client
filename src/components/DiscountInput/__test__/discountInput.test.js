import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import DiscountInput from '..'

describe('DiscountInput', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {discount: 10}
        const errors = {}
        const tree = renderer.create( <DiscountInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {discount: 10}
        const errors = {discount: 'Some error'}
        const tree = renderer.create( <DiscountInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
