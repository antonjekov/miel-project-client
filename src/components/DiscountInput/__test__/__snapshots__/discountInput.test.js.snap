// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountInput matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Discount
  </label>
  <input
    className="form-control form-control is-invalid"
    name="discount"
    onChange={[Function]}
    type="number"
    value={10}
  />
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`DiscountInput matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Discount
  </label>
  <input
    className="form-control form-control is-valid"
    name="discount"
    onChange={[Function]}
    type="number"
    value={10}
  />
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
